file-manager/src/test/scala/it/pagopa/interop/commons/files/S3FileManagerTest.scala

```diff
@@ -0,0 +1,108 @@
+package it.pagopa.interop.commons.files
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.scalatest.concurrent.ScalaFutures
+
+import java.util.concurrent.Executors
+import it.pagopa.interop.commons.files.service.FileManager
+import scala.concurrent.ExecutionContext
+import org.scalatest.time._
+import org.scalatest.BeforeAndAfterAll
+import java.util.concurrent.ExecutorService
+import java.net.URI
+import it.pagopa.interop.commons.files.service.impl.S3ManagerImpl
+import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContextExecutor
+import org.scalatest.BeforeAndAfterEach
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class S3FileManagerTest
+    extends AnyWordSpecLike
+    with Matchers
+    with ScalaFutures
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach {
+
+  override implicit val patienceConfig: PatienceConfig = new PatienceConfig(Span(1, Seconds))
+
+  var fileManager: FileManager              = _
+  var ex: ExecutorService                   = _
+  implicit var ec: ExecutionContextExecutor = _
+
+  override def beforeAll(): Unit = {
+    ex = Executors.newFixedThreadPool(4)
+    ec = ExecutionContext.fromExecutorService(ex)
+    // create locally a minio with a "testBucket"
+    fileManager = new S3ManagerImpl(ec)(
+      _.endpointOverride(URI.create("http://localhost:9000/"))
+        .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("minioadmin", "minioadmin")))
+    )
+  }
+
+  override def afterEach(): Unit = {
+    val deletion: Future[Unit] = for {
+      files <- fileManager.listFiles("testBucket")("")
+      _     <- Future.traverse(files)(fileManager.delete("testBucket"))
+    } yield ()
+    Await.ready(deletion, 1.second)
+  }
+
+  override def afterAll(): Unit = {
+    fileManager.close()
+    ex.shutdown()
+  }
+
+  "S3" ignore {
```

Comment:
These are ignored since they require a local minio installation to simulate s3. I used it though to be sure that the implementation is not unsound.